pkg/prometheus/common.go

```diff
@@ -474,3 +477,36 @@ func BuildStatefulSetService(name string, selector map[string]string, p monitori
 
 	return svc
 }
+
+// This function is responsible for the following:
+//
+// If `ServiceName` is specified, verify that a service with the same name exists and is in the same namespace
```

Comment:
an invariant of the function is that ServiceName isn't nil.
```suggestion
// Verify that the service exists in the Prometheus resource's namespace.
```